lemmy: Add release candidates

adisbladis opened the pull request:
###### Description of changes
- Add RC packages

- Rewrite updater in Python  
When trying to refactor the updater to also support release candidates I found it to become very messy when I wanted reusable sections and accessing data members returned by the Github API.
Python is a more competent programming language that more people in the nix community know how to use.
Additionally it's statically typed now which it wasn't before.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

--- 

cc @matejc @CobaltCause

adisbladis committed:
lemmy: Rewrite updater in Python

When trying to refactor the updater to also support release candidates
I found it to become very messy when I wanted reusable sections
and accessing data members returned by the Github API.

Python is a more competent programming language that more people in
the nix community know how to use.

adisbladis committed:
lemmy: init lemmy-serverUnstable at 0.18.1-rc.10, init lemmy-uiUnstable at 0.18.1-rc.11

happysalada commented:
What do you want to do about the new RC versions coming in, are you happy to test them ?
I'm personally not a fan of the one letter variable names, but actually I don't mind if you are happy to maintain the script. Would you like to become a maintainer ?

adisbladis committed:
lemmy-{server,ui}: add adisbladis as maintainer

adisbladis commented:
> What do you want to do about the new RC versions coming in, are you happy to test them ?

In the immediate term I have show stopper bugs in deploying an instance that are solved by using a release candidate.
I will be testing them on my personal instance.

> I'm personally not a fan of the one letter variable names

Good point. I've refactored the script so the only one-letter variable is now an iteration counter.

> Would you like to become a maintainer ?

Yes! I added myself.

adisbladis commented:
I'm rethinking this now: I think I will package the release candidates out of tree instead.
